### PR TITLE
Remove `expand_depth` in `compile` transform and `max_expansion` in `clifford_t_decomposition` transform

### PR DESCRIPTION
**Pull request by andrijapau**
Context:

Completing the deprecation cycle of `expand_depth` in `compile` transform and `max_expansion` in `clifford_t_decomposition` transform.

Description of the Change:

1. Update deprecations.rst
2. Update changelog-dev.md
3. Remove all deprecated source code

No references found in Catalyst, Lightning, or QML.

[sc-77499]

### CODE REVIEW

**Review by lillian542 — APPROVED** [review]
Looks good to me! 🚀

File: doc/releases/changelog-dev.md [hunk 1]
```diff
@@ -39,6 +39,10 @@
 
 <h3>Breaking changes 💔</h3>
 
+* The ``max_expansion`` argument for :func:`~pennylane.transforms.decompositions.clifford_t_decomposition` has been removed.
+
+* The ``expand_depth`` argument for :func:`~pennylane.transforms.compile` has been removed.
+
```
Comment (mudit2812) [inline doc/releases/changelog-dev.md:45]:
```suggestion
* The `max_expansion` argument for `qml.transforms.clifford_t_decomposition` has been removed.
  [(#6531)](https://github.com/PennyLaneAI/pennylane/pull/6531)

* The `expand_depth` argument for `qml.compile` has been removed.
  [(#6531)](https://github.com/PennyLaneAI/pennylane/pull/6531)

```

File: pennylane/transforms/decompositions/clifford_t_transform.py [hunk 2]
```diff
@@ -373,29 +368,14 @@ def circuit(x, y):
...
-        max_expansion = 6
 
     with QueuingManager.stop_recording():
         # Build the basis set and the pipeline for intial compilation pass
         basis_set = [op.__name__ for op in _PARAMETER_GATES + _CLIFFORD_T_GATES]
         pipelines = [remove_barrier, commute_controlled, cancel_inverses, merge_rotations]
 
         # Compile the tape according to depth provided by the user and expand it
-        with warnings.catch_warnings():
-            warnings.filterwarnings(
```
Comment (mudit2812) [inline pennylane/transforms/decompositions/clifford_t_transform.py:391]: Can you remove `import warnings` from the top of this file or is it used for other things here as well?

**Review by mudit2812 — COMMENTED** [review]
Very minor comment, otherwise LGTM

File: doc/releases/changelog-dev.md (hunk 1 above)
Comment (andrijapau) [reply doc/releases/changelog-dev.md:45]: Good catch! Sorry!

File: pennylane/transforms/decompositions/clifford_t_transform.py (hunk 2 above)
Comment (andrijapau) [reply pennylane/transforms/decompositions/clifford_t_transform.py:391]: It's been used ...
```python
 with warnings.catch_warnings():
        warnings.simplefilter("ignore", RuntimeWarning)
        sd_ops = qml.ops.one_qubit_decomposition(
            qml.matrix(op), op.wires, "ZXZ", return_global_phase=True
        )
```

**Review by mudit2812 — APPROVED** [review]

**Pull request merged**